src/gtk/desktopenv.cpp

```diff
@@ -0,0 +1,50 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/desktopenv.cpp
+// Purpose:     implementation of wxDesktopEnvBase
+// Author:      Igor Korot
+// Created:     2021-09-05
+// Copyright:   (c) 2021 wxWidgets development team
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#include <gio/gio.h>
+#include "wx/translation.h"
+#include "wx/log.h"
+#include "wx/desktopenv.h"
+
+// ============================================================================
+// implementation
+// ============================================================================
+
+/* static */
+bool wxDesktopEnv::MoveToRecycleBin(const wxString &path)
+{
+    bool result = false;
+    if( wxDesktopEnvBase::MoveToRecycleBin( path ) )
+    {
+        GError *error = NULL;
+        GFile *f = g_file_new_for_path( path.c_str() );
+        gboolean res = g_file_trash( f, NULL, &error );
+        if( error && error->code == G_IO_ERROR_NOT_SUPPORTED )
+            wxLogSysError( _( "MoveToRecycleBin: Operation not supported." ) );
```

Comment:
Why do you handle just this error code specifically? You should show the error message if any (i.e. if `error != NULL`) for any error code, use `wxGtkError` class from include/wx/gtk/private/error.h for it.

Reply:
Because "Not supported" is different from any other failure, and the user should know about that specific one.

Reply:
Why is it different? Just give the associated error message, this should be clear enough and it should already say "not supported" in this case.